### PR TITLE
[BISERVER-12883]  Inconsistent behavior with generated/uploaded filen…

### PR DESCRIPTION
**Pull request by alina-ipatina**
…ames

- do not remove extension from files, except files with known extensions

### CODE REVIEW

**Comment by alina-ipatina** [comment]
@mchen-len-son please review.

**Comment by alina-ipatina** [comment]
@mchen-len-son please review.

**Comment by mchen-len-son** [comment]
@mbatchelor @pamval please take a look.
Thank you.

**Comment by mchen-len-son** [comment]
Seems like some tests are non-deterministic - latest wingman's report shows different coverage, but the only change was about fields reordering.

File: extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java [hunk 1]
```diff
@@ -49,10 +50,13 @@
  */
 public class RepositoryFileImportFileHandler implements IPlatformImportHandler {
 
+  private static final List<String> KNOWN_EXTENSIONS = Arrays.asList( "prpt", "prpti", "xdash",
+          "xaction", "xanalyzer", "kjb", "ktr", "xcdf", "xmi", "cda", "xreportspec" );
```
Comment (pamval) [inline extensions/src/main/java/org/pentaho/platform/plugin/services/importer/RepositoryFileImportFileHandler.java]: This should get read from the system configurations, not hardcoded here. Right? I mean, the accepted extensions are defined in importExport.xml or whatever that is. So we need to query the system to know which file extensions are known.

**Comment by alina-ipatina** [comment]
@mchen-len-son @pamval @mbatchelor please review if you have a chance.
Thank you.

**Pull request merged**